File: src/main/java/com/yurticicargo/personnel_task_management/dto/AuthResponse.java
package com.yurticicargo.personnel_task_management.dto;

import com.yurticicargo.personnel_task_management.entity.Role;

public class AuthResponse {

    private String token;
    private Long userId;
    private String username;
    private Role role;

    public AuthResponse(String token, Long userId, String username, Role role) {
        this.token = token;
        this.userId = userId;
        this.username = username;
        this.role = role;
    }

    public String getToken() {
        return token;
    }

    public Long getUserId() {
        return userId;
    }

    public String getUsername() {
        return username;
    }

    public Role getRole() {
        return role;
    }
}

File: mobile/lib/models/auth_response.dart
class AuthResponse {
  final String token;
  final int userId;
  final String username;
  final String role;

  AuthResponse({
    required this.token,
    required this.userId,
    required this.username,
    required this.role,
  });

  factory AuthResponse.fromJson(Map<String, dynamic> json) {
    return AuthResponse(
      token: json['token'],
      userId: json['userId'],
      username: json['username'],
      role: json['role'],
    );
  }
}